frontends/web/src/routes/new-settings/components/advanced-settings/tor-proxy-dialog.tsx

```diff
@@ -0,0 +1,102 @@
+import { Dispatch, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Dialog, DialogButtons } from '../../../../components/dialog/dialog';
+import { Toggle } from '../../../../components/toggle/toggle';
+import { Button, Input } from '../../../../components/forms';
+import { setConfig } from '../../../../utils/config';
+import { socksProxyCheck } from '../../../../api/backend';
+import { alertUser } from '../../../../components/alert/Alert';
+
+type TProps = {
+  open: boolean;
+  config: any;
+  onCloseDialog: () => void;
+  onChangeConfig: (config: any) => void;
+  handleShowRestartMessage: Dispatch<boolean>;
+}
+
+export const TorProxyDialog = ({ open, config, onCloseDialog, onChangeConfig, handleShowRestartMessage }: TProps) => {
+  const [proxyAddress, setProxyAddress] = useState<string>();
+  const [proxyEnabled, setProxyEnabled] = useState<boolean>();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    if (config && config.backend) {
+      setProxyAddress(config.backend.proxy.proxyAddress);
+      setProxyEnabled(config.backend.proxy.useProxy);
+    }
+  }, [config]);
+
+
+  const handleSetProxyButton = async () => {
+    if (!config || proxyAddress === undefined) {
+      return;
+    }
+    const proxy = config.backend.proxy;
+    proxy.proxyAddress = proxyAddress.trim();
+
+    const result = await socksProxyCheck(proxy.proxyAddress);
+    const { success, errorMessage } = result;
+
+    if (success) {
+      setProxyConfig(proxy);
+    } else {
+      alertUser(errorMessage || t('account.fatalError'));
+    }
+  };
+
+  const setProxyConfig = async (proxyConfig: any) => {
+    const config = await setConfig({
+      backend: { proxy: proxyConfig },
+    });
+    setProxyAddress(proxyConfig.proxyAddress);
+    onChangeConfig(config);
+    handleShowRestartMessage(true);
+  };
+
+  const handleToggleProxy = () => {
+    if (!config) {
+      return;
+    }
+    const proxy = { ...config.backend.proxy, useProxy: !proxyEnabled };
+    setProxyConfig(proxy);
```

Comment:
setProxyConfig is async.. shouldn't it do await here or doesn't matter?

I am not sure, old implementation doesn't have proxyEnabled state, the toggle is directly using the config.

Reply:
> setProxyConfig is async.. shouldn't it do await here or doesn't matter?


Ups! Doesnt matter here but better to keep it consistent and have await here, good catch:)

> I am not sure, old implementation doesn't have proxyEnabled state, the toggle is directly using the config.

If im not mistaken, the toggle for proxy is still enabled  using the state (the old settings save the whole config in a state)

<img width="266" alt="Screenshot 2023-05-11 at 02 04 59" src="https://github.com/digitalbitbox/bitbox-wallet-app/assets/28676406/24d3242b-af20-4d2a-b17e-2aee9667ca90">

Reply:
I'll follow the same pattern and remove `proxyEnabled` state now for the new proxy. It's just a redundancy that we don't really need since we're using config state anyways. This config state in the parent page is updated in the dialog component thru this prop `onChangeConfig ` thus we need to keep it :)